lib/src/kvs/ds.rs

```diff
@@ -530,44 +538,65 @@ impl Datastore {
 		Ok(())
 	}
 
-	pub async fn nuke_whole_cluster(&self, tx: &mut Transaction) -> Result<(), Error> {
+	pub async fn clear_unreachable_state(&self, tx: &mut Transaction) -> Result<(), Error> {
 		// Scan nodes
-		let cls = tx.scan_cl(1000).await?;
-		trace!("Found {} nodes", cls.len());
-		for cl in cls {
-			tx.del_cl(
+		let cluster = tx.scan_nd(NO_LIMIT).await?;
+		trace!("Found {} nodes", cluster.len());
+		let mut unreachable_nodes = BTreeMap::new();
+		for cl in &cluster {
+			unreachable_nodes.insert(cl.name.clone(), cl.clone());
+		}
+		// Scan heartbeats
+		let now = tx.clock();
+		let hbs = tx.scan_hb(&now, NO_LIMIT).await?;
+		trace!("Found {} heartbeats", hbs.len());
+		for hb in hbs {
+			unreachable_nodes.remove(&hb.nd.to_string()).unwrap();
+		}
+		// Remove unreachable nodes
+		for (_, cl) in unreachable_nodes {
+			trace!("Removing unreachable node {}", cl.name);
+			tx.del_nd(
 				uuid::Uuid::parse_str(&cl.name).map_err(|e| {
 					Error::Unimplemented(format!("cluster id was not uuid: {:?}", e))
 				})?,
 			)
 			.await?;
 		}
-		// Scan heartbeats
-		let hbs = tx
-			.scan_hb(
-				&Timestamp {
-					value: 0,
-				},
-				1000,
-			)
-			.await?;
-		trace!("Found {} heartbeats", hbs.len());
-		for hb in hbs {
-			tx.del_hb(hb.hb, hb.nd).await?;
+		// Scan node live queries for every node
+		let mut nd_lqs: Vec<Arc<LqValue>> = vec![];
+		for cl in &cluster {
+			let ndlqs = tx.scan_ndlq(&uuid::Uuid::parse_str(&cl.name).map_err(|e| {
+				Error::Unimplemented(format!("cluster id was not uuid when parsing to aggregate cluster live queries: {:?}", e))
+			})?, NO_LIMIT).await?;
+			for ndlq in ndlqs {
+				nd_lqs.push(Arc::new(ndlq));
+			}
 		}
-		// Scan node live queries
-		let ndlqs = tx.scan_ndlq(&self.id, 1000).await?;
-		trace!("Found {} node live queries", ndlqs.len());
-		for ndlq in ndlqs {
-			tx.del_ndlq(&ndlq.nd).await?;
-			// Scan table live queries
-			let tblqs = tx.scan_tblq(&ndlq.ns, &ndlq.db, &ndlq.tb, 1000).await?;
-			trace!("Found {} table live queries", tblqs.len());
-			for tblq in tblqs {
-				tx.del_tblq(&ndlq.ns, &ndlq.db, &ndlq.tb, tblq.lq.0).await?;
+		trace!("Found {} node live queries", nd_lqs.len());
+		// Scan tables for all live queries
+		let mut tb_lqs: Vec<Arc<LqValue>> = vec![];
```

Comment:
<img width="1152" alt="image" src="https://github.com/surrealdb/surrealdb/assets/6785919/0ab5cc12-692f-4f16-ac21-8ee67721d70c">

Reply:
Moved to https://github.com/surrealdb/surrealdb/pull/2510